Shared/Samples/Run valve isolation trace/RunValveIsolationTraceView.swift

```diff
@@ -0,0 +1,111 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+struct RunValveIsolationTraceView: View {
+    /// The view model for the sample.
+    @StateObject var model = Model()
+    
+    var body: some View {
+        MapViewReader { mapViewProxy in
+            MapView(
+                map: model.map,
+                graphicsOverlays: [model.parametersOverlay]
+            )
+            .onSingleTapGesture { screenPoint, mapPoint in
+                model.lastSingleTap = (screenPoint, mapPoint)
+            }
+            .overlay(alignment: .top) {
+                Text(model.statusText)
+                    .frame(maxWidth: .infinity, alignment: .center)
+                    .padding(8)
+                    .background(.ultraThinMaterial, ignoresSafeAreaEdges: .horizontal)
+                    .multilineTextAlignment(.center)
+            }
+            .task {
+                try? await model.setup()
+                await mapViewProxy.setViewpointCenter(model.startingLocationPoint, scale: 3_000)
+            }
+            .task(id: model.lastSingleTap?.mapPoint) {
+                guard let lastSingleTap = model.lastSingleTap else {
+                    return
+                }
+                if let feature = try? await mapViewProxy.identifyLayers(
+                    screenPoint: lastSingleTap.screenPoint,
+                    tolerance: 10
+                ).first?.geoElements.first as? ArcGISFeature {
+                    model.addFilterBarrier(for: feature, at: lastSingleTap.mapPoint)
+                }
+            }
+            .toolbar {
+                ToolbarItemGroup(placement: .bottomBar) {
+                    Toggle("Isolated\nFeatures", isOn: $model.includesIsolatedFeatures)
+                        .toggleStyle(.switch)
+                        .disabled(model.traceCompleted)
+                    Spacer()
+                    Menu(model.selectedCategory?.name ?? "Category") {
+                        ForEach(model.filterBarrierCategories, id: \.self) { category in
+                            Button(category.name) {
+                                model.selectCategory(category)
+                            }
+                        }
+                    }
+                    .disabled(model.traceCompleted)
+                    Spacer()
+                    Button {
+                        if model.traceEnabled {
+                            Task { try await model.trace() }
+                        } else {
+                            model.reset()
+                            Task { await mapViewProxy.setViewpointCenter(model.startingLocationPoint, scale: 3_000) }
+                        }
+                    } label: {
+                        Text(model.traceEnabled ? "Trace" : "Reset")
+                    }
+                    .disabled(!model.traceEnabled && !model.resetEnabled)
```

Comment:
If you wanted, this could be a property on the model called `allowsTraceOrReset` and the implementation can be there.

Reply:
You can have a property on the model that depends on other properties. didSet in the other properties can update a property called `allowsTraceOrReset`.

Reply:
I will leave the disabled modifier condition as-is.